src/taco-livereload/lib/utils/Patcher.js

```diff
@@ -31,63 +28,50 @@ function Patcher(projectRoot, platforms) {
  *    - Removes ATS (App Transport Security) in the case of iOS9+
  *    - Copies the homePage to the mobile app
  *    - Changes the start page of the app to the homePage
- * 
+ *
  * HomePage Role:
- *    - The homePage allows us to ping the BrowserSync server 
+ *    - The homePage allows us to ping the BrowserSync server
  *    - If we can't connect to the BrowserSync server, we display a meaningful error message to the user
  *    - If we can successfully connect to the BrowserSync server, we proceed with loading the app
  */
 Patcher.prototype.patch = function (serverUrl) {
     var self = this;
+    self.removeCSP();
     return promiseUtils.Q_chainmap(self.platforms, function (plat) {
-        var platWWWFolder = multiPlatforms.getPlatformWWWFolder(plat);
-        var platformIndexLocal = path.join(self.projectRoot, platWWWFolder, self.startPage);
-        var cspRemover = new CSPRemover(platformIndexLocal);
-        return cspRemover.Remove().then(function () {
-            var atsRemover = new ATSRemover(self.projectRoot, plat);
-            return atsRemover.Remove();
-        }).then(function () {
-            var platformIndexUrl = url.resolve(serverUrl, path.join(multiPlatforms.getPlatformWWWFolder(plat), self.startPage));
-            return copyHomePage(self.projectRoot, plat, platformIndexUrl);
-        }).then(function (homePage) {
-            return helpers.ChangeStartPage(self.projectRoot, plat, homePage); 
-        });
+        if (plat === 'ios')
+            browserSyncPrimitives.fixATS(self.projectRoot, helpers.GetProjectName(self.projectRoot));
+        var platformIndexUrl = url.resolve(serverUrl, path.join(browserSyncPrimitives.getWWWFolder(plat), self.startPage));
+        var homePage = copyHomePage(self.projectRoot, plat, platformIndexUrl);
+        browserSyncPrimitives.updateConfigXml(self.projectRoot, plat, helpers.GetProjectName(self.projectRoot), homePage);
     });
 };
 
 Patcher.prototype.removeCSP = function () {
-    var self = this;
-    var startPage = helpers.GetStartPage(self.projectRoot);
-
-    var platformWwws = self.platforms.map(function (plat) {
-        return path.join(self.projectRoot, multiPlatforms.getPlatformWWWFolder(plat));
-    });
-
-    return platformWwws.forEach(function (platWWWDir) {
-        var platformIndexLocal = path.join(platWWWDir, startPage);
-        var remover = new CSPRemover(platformIndexLocal);
-        remover.Remove();
-    });
-};
+    this.platforms.forEach(function(plat) {
+        var platWWWFolder = browserSyncPrimitives.getWWWFolder(plat);
+        var platformIndexLocal = path.join(this.projectRoot, platWWWFolder, this.startPage);
+        browserSyncPrimitives.addCSP(platformIndexLocal);
```

Comment:
Is the call to `addCSP` intentional? From the method's name (`removeCSP`), and the previous implementation (`new CSPRemover(platformIndexLocal).Remove();`) I get the feeling this might be a mistake?